src/parser/esprima_cpp/esprima.cpp

```diff
@@ -4758,6 +4758,7 @@ class Parser : public gc {
 
                 if (declarations.size() == 1 && this->matchKeyword(InKeyword)) {
                     RefPtr<VariableDeclaratorNode> decl = declarations[0];
+                    decl->setInitPtr(nullptr);
                     // if (decl->init() && (decl.id.type === Syntax.ArrayPattern || decl.id.type === Syntax.ObjectPattern || this->context->strict)) {
                     if (decl->init() && (decl->id()->type() == ArrayExpression || decl->id()->type() == ObjectExpression || this->context->strict)) {
```

Comment:
After setting the `m_init` to null value, this if statement that checks some syntax errors will be always ignored. Instead of your approach, how about just not generating the bytecode of `m_init` in `ForInStatementNode::generateStatementByteCode`?

Reply:
It can't really be skipped in `ForInStatementNode::generateStatementByteCode`, because the [first time](https://github.com/pando-project/escargot/blob/fa96ce1f18b8cfe897c2c38c4ca5c97465261e09/src/parser/ast/ForInStatementNode.h#L54) it generates the initial value for `m_left`, then the [second time](https://github.com/pando-project/escargot/blob/fa96ce1f18b8cfe897c2c38c4ca5c97465261e09/src/parser/ast/ForInStatementNode.h#L89) it will assign the result of the object enumeration to it (the value that will be used in the current iteration). Both are needed to work correctly, so we can't really skip it here. I'm leaning more towards the suggestion by @robertsipka . I'm currently investigating whether the if check [here](https://github.com/pando-project/escargot/blob/fa96ce1f18b8cfe897c2c38c4ca5c97465261e09/src/parser/ast/VariableDeclarationNode.h#L54) is necessary at all, we may be able to skip that altogether.

Reply:
@pmarkee 
As you commented, generating a store bytecode of `m_left` at the [second time](https://github.com/pando-project/escargot/blob/fa96ce1f18b8cfe897c2c38c4ca5c97465261e09/src/parser/ast/ForInStatementNode.h#L89) would be the proper target point. It may be possible not to generate a store bytecode of m_init in `VariableDeclarationNode::generateStoreByteCode` (instead calling a new function `generateStoreByteCodeWithoutInit`?).